### DISCUSSION TITLE
Sharing syntax highlighting with Neovim config

### DISCUSSION THREAD

**Discussion by psi4j** [sxyazi/yazi#1539] [4. Ideas]
I realize this may be tedious or difficult to implement, but it would be amazing if yazi could be set to use Neovim's Lua syntax highlighting to maintain consistency across both applications. I am now using yazi inside of Neovim as well, so this would create a seamless experience. I figure a lot of individuals using yazi probably also use Neovim, so I thought I'd throw this idea out there in case anyone knows how it could be implemented intelligently without causing too much of a hit on performance.

**Comment by sxyazi** [comment]
You can implement this using [`nvimpager`](https://github.com/lucc/nvimpager), which basically makes your `nvim` work like a pager similar to the `cat` command. Then, you can call it in a custom Yazi previewer. There's even an example of a Yazi previewer here: https://github.com/lucc/nvimpager/issues/101

See docs at https://yazi-rs.github.io/docs/plugins/overview#previewer and https://yazi-rs.github.io/docs/configuration/yazi#plugin.previewers

**Reply by psi4j** [reply]
> You can implement this using [`nvimpager`](https://github.com/lucc/nvimpager), which basically makes your `nvim` work like a pager similar to the `cat` command. Then, you can call it in a custom Yazi previewer. There's even an example of a Yazi previewer here: [lucc/nvimpager#101](https://github.com/lucc/nvimpager/issues/101)
> 
> See docs at https://yazi-rs.github.io/docs/plugins/overview#previewer and https://yazi-rs.github.io/docs/configuration/yazi#plugin.previewers

I knew somebody had to have thought about this before! You're a life saver, my friend. Thank you for making such an awesome and well thought out terminal file manager. This is truly the best one of them all.
